src/format/parsed.rs

```diff
@@ -698,7 +717,13 @@ impl Parsed {
             (_, _, _) => return Err(NOT_ENOUGH),
         };
 
-        if verified { Ok(parsed_date) } else { Err(IMPOSSIBLE) }
+        // verify the quarter if we have a good parsed date so far
+        let verify_quarter = |date: NaiveDate| {
+            let quarter = date.quarter();
+            self.quarter.unwrap_or(quarter) == quarter
+        };
+
+        if verified && verify_quarter(parsed_date) { Ok(parsed_date) } else { Err(IMPOSSIBLE) }
```

Comment:
I'm very new to Rust, and perhaps not super happy with how I added this check. Not sure if there's some better way to do this.

Reply:
Don't think the closure is useful here. Suggest writing it like this:

```rust
if !verified {
    return Err(IMPOSSIBLE);
} else if let Some(parsed) = self.quarter {
    if parsed != parsed_date.quarter() {
        return Err(IMPOSSIBLE);
    }
}

Ok(parsed_date)
```

Reply:
Yeah, made it too complicated. Will fix.